combination.go

```diff
@@ -0,0 +1,125 @@
+package errors
+
+import (
+	"fmt"
+	"io"
+	"strings"
+)
+
+// Combination combines multiple errors into one.
+// The Error method returns the strings from the individual Error methods
+// joined by the new line character '\n'.
+// Combination always has at least one error and returns the first error
+// as result of the Cause method.
+type Combination struct {
```

Comment:
You will notice that the current master API does not share any datatypes except those dealing with stack frames. This is an intentional choice, as typed errors often lead to a lot of issues.

Reply:
Sorry, wasn't aware that this PR was still open, should be closed. My further additions were meant as permanent fork.